addon/usables/use-machine.js

```diff
@@ -0,0 +1,102 @@
+import { interpret } from 'xstate';
+import { setUsableManager } from 'ember-usable';
+import { later, cancel } from '@ember/runloop';
+import { tracked } from '@glimmer/tracking';
+import { getOwner, setOwner } from '@ember/application';
+
+export class InterpreterService {
+  @tracked service;
+  @tracked currentState;
+
+  constructor(machine) {
+    this.machine = machine;
+  }
+
+  get state() {
+    return {
+      currentState: this.currentState,
+      send: this.service.send,
+    };
+  }
+
+  setup() {
+    this.service = interpret(this.machine, {
+      clock: {
+        setTimeout(fn, ms) {
+          return later.call(null, fn, ms);
+        },
+        clearTimeout(timer) {
+          return cancel.call(null, timer);
+        },
+      },
+    })
+      .onTransition((state) => {
+        this.currentState = state;
+      })
+      .start();
+  }
+
+  teardown() {
+    this.service.stop();
+  }
+
+  send() {
+    this.service.send(...arguments);
+  }
+}
+
+export class MachineInterpreterManager {
+  createUsable(context, { machine }) {
+    const owner = getOwner(context);
+
+    const interpreter = new InterpreterService(machine);
```

Comment:
we should check if this is a `Machine` or `MachineConfig` and create a `Machine`, if `machine` is a `MachineConfig` (to eliminate the need for side-effecting code in module-space -- as the XState docs have by default (cause it's easier to teach that way))